packages/flutter_tools/lib/src/android/multidex.dart

```diff
@@ -25,25 +25,32 @@ File _getMultiDexApplicationFile(Directory projectDir) {
 void ensureMultiDexApplicationExists(final Directory projectDir) {
   final File applicationFile = _getMultiDexApplicationFile(projectDir);
   if (applicationFile.existsSync()) {
-    return;
+    if (!applicationFile.readAsStringSync().contains('extends FlutterApplication')) {
```

Comment:
```suggestion
    if (applicationFile.readAsStringSync().contains('android.app.Application')) {
```

Also, it needs a comment about what this case is for.